### PR TITLE
Fix for TLS v1.3 early data mac digest

### PR DESCRIPTION
**Pull request by dgarske**
Fix for TLS v1.3 issue where mac digest changes between early data and server_hello, which can leave section of server_hello response uninitialized. ZD11424

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -7190,13 +7190,8 @@ int DoTls13HandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                                         ENCRYPT_AND_DECRYPT_SIDE, 1)) != 0) {
                     return ret;
                 }
-        #ifdef WOLFSSL_EARLY_DATA
```
Comment (SparkiDev) [inline src/tls13.c:7193]: The Encrypt side is set In SendTls13EndOfEarlyData() after building message.
This is after all the early data has been sent and is the last message that is encrypted with the EarlyData keys.
Setting the Encrypt side early should stop things working.

Checking ssl->earlyData == no_early_data and do both sides is probably the fix.

**Review by dgarske — COMMENTED** [review]
Awesome! Thanks Sean. Approved.

**Comment by guidovranken** [comment]
See my latest comment in ZD 11424 before merging.

**Review by SparkiDev — APPROVED** [review]

**Comment by dgarske** [comment]
Approved. Changes look good. Over to Todd for final. Guido also confirmed. I cannot merge otherwise I would.

**Review by toddouska — APPROVED** [review]

**Pull request merged**